radicle-cli/src/commands/checkout.rs

```diff
@@ -0,0 +1,136 @@
+use std::ffi::OsString;
+use std::path::PathBuf;
+
+use anyhow::anyhow;
+use anyhow::Context as _;
+
+use radicle::prelude::*;
+use radicle::storage::WriteStorage;
+
+use crate::project;
+use crate::terminal as term;
+use crate::terminal::args::{Args, Error, Help};
+
+pub const HELP: Help = Help {
+    name: "checkout",
+    description: "Checkout a radicle project working copy",
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad checkout <id> [<option>...]
+
+Options
+
+    --no-confirm    Don't ask for confirmation during checkout
+    --help          Print help
+"#,
+};
+
+pub struct Options {
+    pub id: Id,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+        use std::str::FromStr;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut id = None;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("no-confirm") => {
+                    // Ignored for now.
+                }
+                Long("help") => return Err(Error::Help.into()),
+                Value(val) if id.is_none() => {
+                    let val = val.to_string_lossy();
+                    let val = Id::from_str(&val).context(format!("invalid id '{}'", val))?;
+
+                    id = Some(val);
+                }
+                _ => return Err(anyhow::anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((
+            Options {
+                id: id.ok_or_else(|| anyhow!("a project id to checkout must be provided"))?,
```

Comment:
nit: This reads better to me 
```suggestion
                id: id.ok_or_else(|| anyhow!("a project id must be provided to checkout"))?,
```